device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -128,18 +129,19 @@
      * RFC 3986 or if the provided {@code connString} is for an x509 authenticated device
      * @throws URISyntaxException if the hostname in the connection string is not a valid URI
      * @throws UnsupportedOperationException if the connection string belongs to a module rather than a device
+     * @deprecated {@link MultiplexingClient} should be used instead of {@link TransportClient} for creating all multiplexed connections.
```

Comment:
So we're marking this constructor as deprecated so users can use the MultiplexingClient instead. But isn't this true only for customers who want to use multiplexed connections? Wouldn't this still be useful to customers who don't want to use multiplexed connections?

Reply:
I'm doing this to impress upon users that we won't be supporting that transport client anymore, and that users who want better multiplexing support should use the multiplexing client. I'm also doing it to cut down on confusion from customers as to which multiplexing client they should use if they are starting from scratch

Reply:
I too am (still) confused by this. If I'm a normal device client user, not multiplexing, will I get any error about this client being deprecated?

Reply:
If you are not multiplexing, then you won't get any errors, no. This tag only deprecates the constructor, not the device client class. This constructor was only used when multiplexing using the now deprecated TransportClient

Reply:
Ah, it wasn't clear to us newbies that `TransportClient` was code for multiplexing.

Reply:
The name of it certainly didn't make it obvious that ```TransportClient``` is for multiplexing. All the more reason to deprecate it